specifications_dbus.md

```diff
@@ -0,0 +1,92 @@
+
+The first argument is the token of the connection, and the second argument is the new endpoint. This method does not return anything.
+
+### org.unifiedpush.Connector1.Unregister (String) → nothing
+
+The distributor calls this method to inform the connector that it has been unregistered for some server-side or distributor-side reason. If the app wants to remain registered, it SHOULD simply call [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) again after getting unregistered. A response indicating "REGISTRATION_REFUSED" would mean that the app is no longer allowed to register.
+
+The argument of the method is the token of the connection. This method does not return anything.
+
+## Distributor
+
+A *distributor* is a service which runs on the device, receiving push messages via whatever transport it wants, and sends them on to the connector using the connector API.
+
+## Distributor API
+
+Distributors MUST provide a service with a name beginning with `org.unifiedpush.Distributor.` followed by the name of the distributor (for example, `org.unifiedpush.Distributor.distributorname` where `distributorname` is the distributor's name)
+
+The distributor MUST implement the `org.unifiedpush.Distributor1` interface at the object path `/org/unifiedpush/Distributor`.
+
+### org.unifiedpush.Distributor1.Register (String, String) → (String, String)
+
+The connector calls this method to register for push messages, or to retreive the newest push endpoint (as it may change at any time). See also the `org.unifiedpush.Connector1.NewEndpoint` method.
+
+When it has not registered before, the connector generates a random string to use as its *token* and calls this method with the token as its argument to register. At every subsequent startup of the app, it SHOULD call this method with the same token as before, to fetch the newest endpoint from the connector.
+
+The method returns two strings. The first string MUST be either "NEW_ENDPOINT", "REGISTRATION_REFUSED", or "REGISTRATION_FAILED". It is "NEW_ENDPOINT" if registration succeeded. It is "REGISTRATION_REFUSED" in case the registration attempt was refused, and the connector SHOULD NOT try to register again. It is "REGISTRATION_FAILED" in case the registration failed for some other reason, and the connector MAY try to register again. The second string MAY contain a reason string, or it may be empty.
+
+If registration succeeded, the distributor MUST call the connector's [org.unifiedpush.Connector1.NewEndpoint](#orgunifiedpushconnector1newendpoint-string-string--nothing) method to deliver the new push endpoint to it.
+
+### org.unifiedpush.Distributor1.Unregister (String) → nothing
+
+The connector calls this method to unregister for push messages, with the connection token as its argument. The method does not return anything.
```

Comment:
MUST call
with the following argument:
* token: the token supplied by the end user application during registration

Reply:
Shouldn't a "MUST call" be paired with a situation when the connector must call it? After all, it doesn't ever have to call this method if it never has a reason to want to unregister.

Should it be along the lines of "MUST call [...] when it wants to unregister from receiving push messages"?